libs/community/langchain_community/document_loaders/gitbook.py

```diff
@@ -10,7 +12,7 @@ class GitbookLoader(WebBaseLoader):
     """Load `GitBook` data.
 
     1. load from either a single page, or
-    2. load all (relative) paths in the navbar.
+    2. load all (relative) paths in the sitemap, handling nested sitemap indexes.
```

Comment:
The implementation seems too complex to me.

Why does GitbookLoader not inherit from DocumentLoader?

It can implement lazy_load and alazy_load and use WebBaseLoader as it needs to fetch content?